web/cobrands/fixmystreet/offline_report.js

```diff
@@ -200,32 +301,92 @@ fixmystreet.offlineReporting = (function() {
         dropzone.options.maxFiles -= 1;
     }
 
-    function deleteDrafts() {
-        return idbKeyval.set('draftOfflineReports', []).then(function() {
-            return restoreDraft();
+    function deleteDraft(i) {
+        loadDrafts().then(function(drafts) {
+            drafts.splice(i, 1);
+            return idbKeyval.set('draftOfflineReports', drafts).then(function() {
+                resetDraftForm();
+                updateDraftList();
+            });
         });
     }
 
+    function setCurrentDraftID(draft_id) {
+        return idbKeyval.set('currentOfflineDraftID', draft_id);
+    }
+
+    // Wraps a function to ensure it's only called at most once every
+    // <limit> milliseconds. We use it here to limit the per-keystroke draft
+    // saving which is a bit slow on some mobile devices. Once per second is
+    // reasonable enough.
+    function throttle(fn, limit) {
+        var wait = false;
+        var that, args;
+
+        var throttled = function() {
+            if (args == null) {
+                wait = false;
+            } else {
+                fn.apply(that, args);
+                args = null;
+                setTimeout(throttled, limit);
+            }
+        };
+
+        return function () {
+            if (wait) {
+                that = this;
+                args = arguments;
+            } else {
+                fn.apply(that, args);
+                wait = true;
+                setTimeout(throttled, limit);
+            }
+        };
+    }
+
     return {
         offlineFormSetup: function() {
             dropzoneSetup();
-            $(".js-delete-drafts").on("click", function(e) {
+
+            $(document).on('click', '.js-continue-draft', function() {
+                var id = parseInt(this.parentNode.parentNode.getAttribute('data-id'), 10);
+                loadDraft(id).then(function(d) {
+                    location.href = "/report/new?restoreDraft=" + id + "&latitude=" + d.latitude + "&longitude=" + d.longitude;
```

Comment:
This happens at present, but just noticed now in testing - if you haven't saved a location for a report, then this button redirects to the front page, and even if you provide a location your draft text is lost (as restoreDraft not there any more). Worth opening a new ticket for rather than fixing here, I think, eg the reporting moving pin bug is more urgent.